shlib/tests/lib_tests/assertions_tests/assert_equal_test.sh

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env bash
+
+# --- begin runfiles.bash initialization v2 ---
+# Copy-pasted from the Bazel Bash runfiles library v2.
+set -uo pipefail; f=bazel_tools/tools/bash/runfiles/runfiles.bash
```

Comment:
I know this is copy-paste but my OCD doesn't allow me to leave this alone.

```suggestion
set -o nounset -o pipefail; f=bazel_tools/tools/bash/runfiles/runfiles.bash
```

Reply:
Does it violate our style guide for shell code?

Reply:
I went ahead and changed this. I do want to note that we are not consistent in this repo about using explicit Bash options. If that is what we want, we should add it to a style guide. [Counter example](https://github.com/aspect-build/bazel-lib/blob/main/lib/paths.bzl#L16)

Reply:
Yes, how about we start a style guide for Bash? It's a language that can make lots of programmers mad because they don't understand the code. If we aim for rules that make the code more intelligible, maybe we can save some lost souls.